Shared/Samples/Generate offline map with local basemap/GenerateOfflineMapWithLocalBasemapView.swift

```diff
@@ -0,0 +1,294 @@
+                                    isGeneratingOfflineMap = true
+                                }
+                                Button("Use Online Basemap") {
+                                    model.usesLocalBasemap = false
+                                    isGeneratingOfflineMap = true
+                                }
+                            }
+                            .task(id: isGeneratingOfflineMap) {
+                                // Ensures generating an offline map is true.
+                                guard isGeneratingOfflineMap else { return }
+                                
+                                // Creates a rectangle from the area of interest.
+                                let viewRect = geometry.frame(in: .local).inset(
+                                    by: UIEdgeInsets(
+                                        top: 20,
+                                        left: geometry.safeAreaInsets.leading + 20,
+                                        bottom: 44,
+                                        right: -geometry.safeAreaInsets.trailing + 20
+                                    )
+                                )
+                                
+                                // Creates an envelope from the rectangle.
+                                guard let extent = mapView.envelope(fromViewRect: viewRect) else { return }
+                                
+                                do {
+                                    // Generates an offline map.
+                                    try await model.generateOfflineMap(extent: extent)
+                                } catch {
+                                    self.error = error
+                                }
+                                
+                                // Sets generating an offline map to false.
+                                isGeneratingOfflineMap = false
+                            }
+                        }
+                    }
+            }
+        }
+    }
+}
+
+private extension GenerateOfflineMapWithLocalBasemapView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        /// The offline map that is generated.
+        @Published private(set) var offlineMap: Map!
+        
+        /// A Boolean value indicating whether the generate button is disabled.
+        @Published private(set) var isGenerateDisabled = true
+        
+        /// A Boolean value indicating if the local basemap is available to be used.
+        @Published var canUseLocalBasemap: Bool = {
+            FileManager.default.fileExists(atPath: URL.napervilleImageryBasemap.path())
+        }()
+        
+        /// A Boolean value indicating whether the offline map uses the local or online basemap.
+        @Published var usesLocalBasemap = false
```

Comment:
Do these two properties need to be `Published`? Not sure, but they don't seem to drive any state.